src/CoreExtensions/Navigation/Builder.php

```diff
@@ -71,7 +71,7 @@ protected function getChildren(Document $parentDocument): array
         foreach ($children as $child) {
             $permissionResource = $child->getProperty('permission_resource');
 
-            if (empty($permissionResource) || $this->currentUser && $this->service->isAllowed($this->currentUser, $child->getProperty('permission_resource'))) {
+            if (empty($permissionResource) || $this->service->isAllowed($this->currentUser, $child->getProperty('permission_resource'))) {
```

Comment:
I think $this->currentUser can be null. Change the PhpDoc for the attribute.

Reply:
Thanks! Changed it :)